Tools/PIDML/KaonPidTask.cxx

```diff
@@ -0,0 +1,139 @@
+  Configurable<bool> cfgUseCCDB{"useCCDB", false, "Whether to autofetch ML model from CCDB. If false, local file will be used."};
+
+  o2::ccdb::CcdbApi ccdbApi;
+
+  Filter collisionFilter = (nabs(aod::collision::posZ) < cfgZvtxCut);
+  Filter trackFilter = (nabs(aod::track::eta) < cfgEtaCut) && (aod::track::pt > cfgMinPtCut) && (aod::track::pt < cfgMaxPtCut);
+
+  // Applying filters
+  using MyFilteredCollisions = soa::Filtered<o2::aod::MyCollisions>;
+  using MyFilteredCollision = MyFilteredCollisions::iterator;
+
+  Partition<o2::aod::MyTracks> positive = (nabs(aod::track::eta) < cfgEtaCut) && (aod::track::pt > cfgMinPtCut) && (aod::track::pt < cfgMaxPtCut) && (aod::track::signed1Pt > cfgChargeCut);
+  Partition<o2::aod::MyTracks> negative = (nabs(aod::track::eta) < cfgEtaCut) && (aod::track::pt > cfgMinPtCut) && (aod::track::pt < cfgMaxPtCut) && (aod::track::signed1Pt < cfgChargeCut);
+
+  void init(o2::framework::InitContext&)
+  {
+    AxisSpec vtxZAxis = {100, -20, 20};
+    std::vector<double> ptBinning = {0.5, 0.6, 0.7, 0.8, 0.9, 1.0, 1.1, 1.2, 1.3, 1.4, 1.5, 1.6, 1.8, 2.0, 2.2, 2.4, 2.8, 3.2, 3.6, 4.};
+    AxisSpec ptAxis = {ptBinning, "#it{p}_{T} (GeV/#it{c})"};
+
+    if (cfgUseCCDB) {
+      ccdbApi.init(cfgCCDBURL); // Initializes ccdbApi when cfgUseCCDB is set to 'true'
+    }
+    pidModel = std::make_shared<PidONNXModel>(cfgPathLocal.value, cfgPathCCDB.value, cfgUseCCDB.value, ccdbApi, cfgTimestamp.value, cfgPid.value, static_cast<PidMLDetector>(cfgDetector.value), cfgCertainty.value);
+
+    histos.add("hChargePos", ";z;", kTH1F, {{3, -1.5, 1.5}});
+    histos.add("hChargeNeg", ";z;", kTH1F, {{3, -1.5, 1.5}});
+    histos.add("hInvariantMass", ";M_{k^{+}k^{-}} (GeV/#it{c}^{2});", kTH1F, {{100, 0., 2.}});
+    histos.add("hdEdXvsMomentum", ";P_{K^{+}K^{-}}; dE/dx in TPC (keV/cm)", kTH2F, {{100, 0., 4.}, {200, 20., 400.}});
+  }
+
+  void process(MyFilteredCollision const& coll, o2::aod::MyTracks const& tracks)
+  {
+    auto groupPositive = positive->sliceByCached(aod::track::collisionId, coll.globalIndex());
+    auto groupNegative = negative->sliceByCached(aod::track::collisionId, coll.globalIndex());
+    histos.fill(HIST("hZvtx"), coll.posZ());
+
+    for (auto track : groupPositive) {
+      histos.fill(HIST("hChargePos"), track.sign());
+      if (pidModel.get()->applyModelBoolean(track)) {
+        histos.fill(HIST("hdEdXvsMomentum"), track.p(), track.tpcSignal());
+      }
+    }
+
+    for (auto track : groupNegative) {
+      histos.fill(HIST("hChargeNeg"), track.sign());
+      if (pidModel.get()->applyModelBoolean(track)) {
+        histos.fill(HIST("hdEdXvsMomentum"), track.p(), track.tpcSignal());
+      }
+    }
+
+    for (auto& [pos, neg] : combinations(soa::CombinationsFullIndexPolicy(groupPositive, groupNegative))) {
+      if (!(pidModel.get()->applyModelBoolean(pos))) {
+        continue;
+      } else if (!(pidModel.get()->applyModelBoolean(neg))) {
```

Comment:
`else if` doesn't make sense after `continue`. It should be just `if`. Alternatively, you could put both conditions for `pos` and `neg` into one OR-ed `if` (with `||`)


It is about programming design and it doesn't impact the work of your task, so I accept but please correct that in your next PR.